src/ringmesh/io/geomodel_builder_resqml.cpp

```diff
@@ -0,0 +1,968 @@
+        GeoModelBuilderRESQML& builder, GeoModel3D& geomodel )
+        : builder_( builder ), geomodel_( geomodel )
+    {
+    }
+
+    void GeoModelBuilderRESQMLImpl::deserialize( EpcDocument& pck )
+    {
+        std::string resqmlResult = pck.deserialize();
+        if( !resqmlResult.empty() )
+        {
+            std::cerr << resqmlResult << std::endl;
+            std::cout << "Press enter to continue..." << std::endl;
+            std::cin.get();
+        }
+
+        std::cout << "EpcDocument name " << pck.getName() << " in "
+                  << ( pck.getStorageDirectory().empty()
+                             ? "working directory."
+                             : pck.getStorageDirectory() )
+                  << std::endl;
+
+        unsigned int hdfProxyCount = pck.getHdfProxyCount();
+        std::cout << "There are " << pck.getHdfProxyCount()
+                  << " hdf files associated to this epc document." << std::endl;
+        for( unsigned int hdfProxyIndex = 0; hdfProxyIndex < hdfProxyCount;
+             ++hdfProxyIndex )
+        {
+            std::cout << "Hdf file relative path : "
+                      << pck.getHdfProxy( hdfProxyIndex )->getRelativePath()
+                      << std::endl;
+        }
+        for( size_t warningIndex = 0; warningIndex < pck.getWarnings().size();
+             ++warningIndex )
+        {
+            std::cout << "Warning #" << warningIndex << " : "
+                      << pck.getWarnings()[warningIndex] << std::endl;
+        }
+    }
+
+    namespace
+    {
+        RELATION get_contact_mode(
+            gsoap_resqml2_0_1::resqml2__ContactMode mode )
+        {
+            switch( mode )
+            {
+            case gsoap_resqml2_0_1::resqml2__ContactMode__baselap:
+                return RELATION::BASELAP;
+            case gsoap_resqml2_0_1::resqml2__ContactMode__erosion:
+                return RELATION::ERODED;
+            case gsoap_resqml2_0_1::resqml2__ContactMode__proportional:
+                return RELATION::CONFORMABLE;
+            default:
+                ringmesh_assert_not_reached;
+                return RELATION::CONFORMABLE;
+            }
+        }
+    } // namespace
+
+    gmge_id GeoModelBuilderRESQMLImpl::find_layer(
+        const Region3D& region ) const
+    {
+        std::set< const Interface3D* > horizons;
+        for( auto b : range( region.nb_boundaries() ) )
+        {
+            const Surface3D& surface = region.boundary( b );
+            const Interface3D& interface = static_cast< const Interface3D& >(
+                surface.parent( Interface3D::type_name_static() ) );
+            if( GMGE::is_stratigraphic_limit( interface.geological_feature() ) )
+            {
+                horizons.insert( &interface );
+            }
+        }
+
+        if( horizons.size() >= 1 && horizons.size() <= 2 )
+        {
+            // TODO the following detection is not working if the region doesn't
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1135)